### DISCUSSION TITLE
How to group download tasks during the curl download process and perform and speed limit them respectively?

### DISCUSSION THREAD

**Discussion by wzssga** [curl/curl#12336] [Q&A] [answered]
How to group download tasks during the curl download process and perform and speed limit them respectively?
The only way I can think of is to enable multiple CURLM* multi at the same time, and then add tasks and drivers respectively.

1. Is it possible to initialize multiple multi instances in a single Curl and use them to download different types of tasks?

The implementation process is as follows：
CURLM* _multi_1 = curl_multi_init();
CURLM* _multi_2 = curl_multi_init();

//insert _multi_1
curl_multi_remove_handle(_multi_1, handle1);
curl_multi_remove_handle(_multi_1, handle2);
//insert _multi_2
curl_multi_remove_handle(_multi_2, handle3);
curl_multi_remove_handle(_multi_2, handle4);

//perform each
curl_multi_perform(_multi_1,n)
curl_multi_perform(_multi_2,m)

2. Is there any other better way to group tasks and drive and speed limit them separately?

**Comment by bagder** [comment, marked as answer]
> Is it possible to initialize multiple multi instances in a single Curl and use them to download different types of tasks?

Sure

> Is there any other better way to group tasks and drive and speed limit them separately?

There is no speed limit in libcurl on a per-multi basis, only per-transfer basis.

**Reply by wzssga** [reply]
Thanks for your answer, it helps me a lot.

I will use multiple multi objects to manage different types of downloads, and I will further verify it later.

thanks again!
